src/libmongoc/doc/mongoc_client_new.rst

```diff
@@ -21,9 +21,10 @@ Parameters
 Returns
 -------
 
-A newly allocated :symbol:`mongoc_client_t` if the URI parsed successfully, otherwise ``NULL``.
+A newly allocated :symbol:`mongoc_client_t` that should be freed with :symbol:`mongoc_client_destroy()` when no longer in use. On error, ``NULL`` is returned and an error or warning will be logged.
```

Comment:
Added "or warning" here for the case where a URI error is emitted. See https://github.com/mongodb/mongo-c-driver/pull/921/files#r775994888 for context.